fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonUtils.xtend

```diff
@@ -424,13 +429,18 @@ class PythonUtils {
 		    «protectedArea(cls, attribute.name + "_read", attribute.setAttrVal, false)»
 		    
     '''
-        
-    def readAttributeMethodHL(PogoDeviceClass cls, Attribute attribute) '''
-        def read_«attribute.name»(self):
+    def readAttributeMethodHL(PogoDeviceClass cls, Attribute attribute, boolean isDynamic) '''
+        def read_«attribute.name»(self«IF isDynamic», attr«ENDIF»):
                 «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
                 «openProtectedAreaHL(cls, attribute.name + "_read")»
+                «IF isDynamic==false»
                 """Return the «attribute.name» attribute."""
                 return self._«attribute.pythonAttributeVariableNameHL»
+                «ELSE»
+                """Return the «attribute.name»_read attribute."""
+                attr.set_value(self.attr_«attribute.name»_read)
```

Comment:
If we are going to include a real implementation that is active on first generation, then the variable we are using, `self.attr_«attribute.name»_read`, must exist.   I see that the initialisation of the variable, created by by [`dynamicAttributeDefaultValueExample`](https://github.com/tango-controls/pogo/blob/410f63460b632a86963bacc738d523fe2591d1d6/fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonDeviceHL.xtend#L358), will be inside a docstring, so the variable won't exist.  

Either put the implementations for read and write inside docstrings, or take the example initialisation out of the docstrings.  I think keeping everything in docstrings would be more consistent with the previous behaviour of Pogo.

Reply:
Thx,
Ok so to summarize our review:
- in the next task I will apply the changes to remove the _read suffix
- also the current approach to providing the hint on how to implement the dynamic attributes isn't perfect and has a problem as you describe in your comment. Solving this in the pogo implementation will be required an additional method to check the code inside of the comments (```code```) which should be generated only once. And this isn't trivial. What is your proposition about this problem?

Also getting back to current PR, what are your opinion about the current state of this PR, when the code will be ready to be merged in your opinion, and what I should do in this PR? Because I meet the requirements and provide the solution to the problem in the ticket. I follow the current way that we have in the pogo.

Currently what left to do is to get back to the previous solution to add the suggestion about the implementation of the dynamic attributes in the comment block (if this is accepted way by you).

Reply:
> Solving this in the pogo implementation will be required an additional method to check the code inside of the comments (code) which should be generated only once. And this isn't trivial. What is your proposition about this problem?

The problem is that the initialisation code is a commented-out (docstring) example, while the read/write implementations are not commented out and use an undefined variable.  My proposition is not to do any extra checking - you are right that it wouldn't be trivial.   Option 1:  generate code that isn't commented out and works in both the initialiser and the read/write implementations.  Option 2:  generate code that is commented out in both the initialiser and the read/write implementations.

> Also getting back to current PR, what are your opinion about the current state of this PR, when the code will be ready to be merged in your opinion, and what I should do in this PR? Because I meet the requirements and provide the solution to the problem in the ticket. I follow the current way that we have in the pogo.

I think it has an issue that need to be fixed before it can be merged.  It doesn't meet the requirements because (I think) the generated code will raise an exception if a client tries to read a dynamic attribute before writing to it.  Have you tested the generated code, and tried to read an attribute before writing it?

> Currently what left to do is to get back to the previous solution to add the suggestion about the implementation of the dynamic attributes in the comment block (if this is accepted way by you).

In the current PR, I think the easiest so solution is to leave the code where it is, but put it inside a docstring, so that it will not be executed (and raise an exception).  See my suggestions in the additional comments.